documentation/source/runner.rst

```diff
@@ -0,0 +1,80 @@
+.. _howto-python-runner:
+
+******************************
+Building HDL and Running Tests
+******************************
+
+.. versionadded:: 1.7
+
+.. warning::
+    Python runners and associated APIs are an experimental feature and subject to change.
+
+
+The Python Test Runner (short: "runner") described here is a replacement
+for cocotb's traditional Makefile flow.
+It builds the HDL for the simulator and runs the tests.
+
```

Comment:
Maybe we should also say here what it is not: a tool that will be able to handle any "strange situation". (Point to "Custom Flows" for such purposes.)